src/App.tsx

```diff
@@ -1,25 +1,146 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { UsersSelect } from './components/UsersSelect/UsersSelect';
+
+type HandleChangeTypes = HTMLInputElement | HTMLSelectElement;
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const allTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(allTodos);
+  const [inputValue, setInputValue] = useState('');
+  const [selectValue, setSelectValue] = useState(0);
+  const [inputValidation, setInputValidation] = useState(true);
+  const [selectValidation, setSelectValidation] = useState(true);
+
+  const handleChange = (event: ChangeEvent<HandleChangeTypes>) => {
+    const element = event.target;
+
+    switch (element.name) {
+      case 'title':
+        setInputValue(element.value);
+        setInputValidation(true);
+        break;
+
+      case 'select':
+        setSelectValue(+element.value);
+        setSelectValidation(true);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const isFormValid = () => {
+    let isValid = true;
+
+    if (inputValue === '') {
+      setInputValidation(false);
+      isValid = false;
+    }
+
+    if (selectValue === 0) {
+      setSelectValidation(false);
+      isValid = false;
+    }
+
+    return isValid;
+  };
+
+  const handleFormSubmit = (e: FormEvent): void => {
+    e.preventDefault();
+    if (!isFormValid()) {
+      return;
+    }
+    const todosId = todos.map(todo => todo.id);
+    const largestId = Math.max(...todosId);
+    const user = getUser(selectValue);
+
+    const newTodo = {
+      id: largestId + 1,
+      title: inputValue,
+      userId: selectValue,
+      user,
+      completed: false,
+    };
+
+    setTodos(current => ([
+      ...current,
+      newTodo,
+    ]));
+
+    setInputValue('');
+    setSelectValue(0);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(e) => handleFormSubmit(e)}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleField">
+            Title:
```

Comment:
Better will be to give a wrapper here
div or span
```suggestion
          <div>
            Title:
          </div>
```